### PR TITLE
LPS-187441 - Automation  ViewChildPages  

### PR DESCRIPTION
**Pull request by marcosvictor22**
@david-gutierrez-mesa

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by marcosvictor22** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 1]
```diff
@@ -11,6 +11,10 @@ definition {
 	var siteURLKey = StringUtil.toLowerCase(${randomSiteName});
 
 	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:17]: Please don't add these kinds of steps. In our case, we don't set up virtual instance.

**Review by Tim-Cao — COMMENTED** [review]
@marcosvictor22 Could you refactor according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 2]
```diff
@@ -1655,6 +1659,60 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-187441. Only the active page can show the child pages."
+	@priority = 4
+	test ViewChildPages {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1664]: Move case to PagesTree.testcase then rename to ShowChildPages

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 3]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+		task ("Given that the user created a Page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1665]: Given a user has pages with child pages

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 4]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+		task ("When a user creates a ChildPage from the Page") {
+			JSONLayout.addPublicLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Child Page Name 1",
+				parentLayoutName = "Test Page Name1");
+
+			JSONLayout.addPublicLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Child Page Name 2",
+				parentLayoutName = "Test Page Name2");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1684]: Move to Given block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 5]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+			JSONLayout.addPublicLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Test Page Name1");
+
+			JSONLayout.addPublicLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Test Page Name2");
+		}
+
+		task ("When a user creates a ChildPage from the Page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1675]: When the user shows child pages in Page Tree

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 6]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+				parentLayoutName = "Test Page Name1");
+
+			JSONLayout.addPublicLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Child Page Name 2",
+				parentLayoutName = "Test Page Name2");
+
+			PagesAdmin.openPagesAdmin(siteURLKey = ${siteURLKey});
+
+			PagesAdmin.expandPageTree(depth = 2);
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1688]: Portlet.expandTree();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 7]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+				groupName = ${randomSiteName},
+				layoutName = "Child Page Name 1",
+				parentLayoutName = "Test Page Name1");
+
+			JSONLayout.addPublicLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Child Page Name 2",
+				parentLayoutName = "Test Page Name2");
+
+			PagesAdmin.openPagesAdmin(siteURLKey = ${siteURLKey});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1686]: PagesFinder.goToPagesFinder();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 8]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+
+			PagesAdmin.expandPageTree(depth = 2);
+
+			MouseOver(
+				key_itemName = "Test Page Name2",
+				locator1 = "PagesAdmin#LIST_GROUP_ITEM");
+
+			AssertVisible(
+				key_pageName = "Test Page Name2",
+				locator1 = "PagesAdmin#LIST_GROUP_ITEM_PLUS_ICON");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1706]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 9]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+		}
+
+		task ("Then check if the Child Page is present on the home") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1709]: Then the user sees child pages in Page Tree

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 10]
```diff
@@ -1655,6 +1659,60 @@ definition {
...
+			PagesFinder.goToPagesFinder();
+
+			PagesFinder.selectPageType(optionValue = "Pages Hierarchy");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1712]: PagesFinder.viewPageNameInPagesFinder

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase (hunk 1 above)
Comment (marcosvictor22) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:17]: Sorry, I put it to test and ended up forgetting to take it off!

**Comment by marcosvictor22** [comment]
@Tim-Cao Done!

**Comment by Tim-Cao** [comment]
@marcosvictor22 See changes https://github.com/liferay-echo/liferay-portal/pull/12836

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**